src/graphics/weather.c

```diff
@@ -0,0 +1,360 @@
+#include "weather.h"
+
+#include "core/config.h"
+#include "core/dir.h"
+#include "core/file.h"
+#include "core/random.h"
+#include "graphics/color.h"
+#include "graphics/graphics.h"
+#include "graphics/screen.h"
+#include "scenario/property.h"
+#include "sound/speech.h"
+#include "time.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <math.h>
+
+#define DRIFT_DIRECTION_RIGHT 1
+#define DRIFT_DIRECTION_LEFT -1
+
+typedef struct {
+    float x;
+    float y;
```

Comment:
From what I see in the code, there are many `float`s we can do without by adapting the calculations to use `int`s.

Please do that. By design we still avoid using `float`s in the code as much as possible.

From my understanding of the code, all `float`s can be replaced by `int`s with little to no accuracy loss.